app/services/common/base.rb

```diff
@@ -107,7 +107,7 @@ def batch_tag
 
       barcodes = batch.relationships.dig('barcodes', 'data')&.map { |label| label['id'] }
 
-      matches = barcodes&.select { |label| /[A-Z0-9]{24,}/.match?(label) }
+      matches = barcodes&.select { |label| /[A-Z0-9]{24,24}/.match?(label) }
```

Comment:
we should move this regexp into a `MetrcUtils.looks_like_metrc_tag?` or something like that